### ISSUE TITLE
Resource: nginx_install 

### ISSUE THREAD

**Issue by damacus** [sous-chefs/nginx#486]
# Brief Description 🗣 

Add a `nginx_install` resource

## Scenario 🎭 

Install Nginx in the standard pattern. (Install/configure resources)

```text
As a cookbook user
And as a cookbook maintainer
I want to install nginx in the same way as I do HAProxy or Apache2
So that I don't have to learn another pattern for installing software using Chef
```

```text
As a maintainer of the Nginx cookbook
I want to be able to easily understand how to install Nginx
So that I can maintain the cookbook with less effort
```

## 🔍 Acceptance Criteria 🔎 

- [ ] Install nginx via package resource
- [ ] Adds any repositories required for a recent version of Nginx
- [ ] Current recipe install methods call the custom resource
- [ ] Does not configure Nginx
- [ ] Unit tests for all helper methods

**Comment by fernandoaleman** [comment]
@damacus I'd be interested in starting this. We use `nginx` all the time with our `rails` apps.

**Comment by fernandoaleman** [comment]
@damacus Hi Dan. For the `nginx` upgrade, do we want to keep support for `centos-6` and `amazonlinux` or drop them like in `apache2`? Also, I'll go ahead and add support for `amazonlinux-2`.

**Comment by damacus** [comment]
If we're doing *anything* with a service file we should only support systemd now. Unless someone else chimes in with *I still use a platform released 9 years ago*

**Comment by damacus** [comment]
Closed via #492
